usecases-integration/src/test/java/eu/europeana/cloud/integration/usecases/ActiveRecordsTestCase.java

```diff
@@ -57,13 +68,35 @@ public void executeTestCase() throws CloudException, MCSException, IOException {
             List<CloudIdAndTimestampResponse> intersectedCloudIdAndTimestamps = intersectCloudIdAndTimestampResponsesBasedOnCloudId(dereferenceCloudIdAndTimestampResponseList, publishedCloudIdAndTimestampResponseList);
             assertNotNull(intersectedCloudIdAndTimestamps);
             assertEquals(intersectedCloudIdAndTimestamps.size(), 1);
+
+            String utcFormateDateString = getUTCDateString(intersectedCloudIdAndTimestamps);
+            RepresentationRevisionResponse representationRevisionResponse = sourceRecordServiceClient.getRepresentationRevision(intersectedCloudIdAndTimestamps.get(0).getCloudId(), SOURCE_REPRESENTATION_NAME, DEREFERENCE_REVISION, SOURCE_PROVIDER_ID, utcFormateDateString);
+            assertNotNull(representationRevisionResponse.getFiles());
+            assertEquals(representationRevisionResponse.getFiles().size(), 1);
+
+            InputStream stream = sourceFileServiceClient.getFile(representationRevisionResponse.getCloudId(), SOURCE_REPRESENTATION_NAME, representationRevisionResponse.getVersion(), representationRevisionResponse.getFiles().get(0).getFileName());
+            List<String> lines = IOUtils.readLines(stream);
+            assertNotNull(lines);
+            assertEquals(lines.size(), 1);
+            assertEquals(lines.get(0), FILE_CONTENT);
+
+            //Do any Conversion to the file content; Mock it in here to use the same content
+            String convertedContent = lines.get(0);
+            String uri = sourceDatasetHelper.addFileToNewRepresentation(SOURCE_REPRESENTATION_NAME, SOURCE_PROVIDER_ID, convertedContent);
+            assertNotNull(uri);
             System.out.println("ActiveRecordsTestCase Finished Successfully ..");
 
         } finally {
             cleanUp();
         }
     }
 
+    private String getUTCDateString(List<CloudIdAndTimestampResponse> intersectedCloudIdAndTimestamps) {
```

Comment:
I do not see any reason to pass the whole list of CloudIdAndTimestampResponse objects if you use only the timestamp from the first element.